src/Service.Tests/SqlTests/GraphQLQueryTests/GraphQLQueryTestBase.cs

```diff
@@ -1725,6 +1725,333 @@ query queryAccountAndParent{
+
+            string expected = await GetDatabaseResultAsync(msSqlQuery);
+            JsonDocument expectedDocument = JsonDocument.Parse(expected);
+            JsonElement expectedArray = expectedDocument.RootElement;
+            SqlTestHelper.AssertNumericAggregations(groupByArray, expectedArray, isfieldsPresentInResponse: false);
+        }
+
+        /// <summary>
+        /// Test to check GraphQL support for sum aggregation with aliases.
+        /// This test verifies that the SQL query results are correctly mapped to the expected GraphQL format.
+        /// </summary>
+        [TestMethod]
+        public async Task TestSupportForHavingAggregation(string msSqlQuery)
+        {
+            string graphQLQueryName = "stocks_prices";
+            string graphQLQuery = @"
+    {
+        stocks_prices {
+            groupBy {
+                aggregations {
+                    sum_price: sum(field: price, having:{ gt: 50 })
+                }
+            }
+        }
+    }";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLQuery, graphQLQueryName, isAuthenticated: false);
+            JsonElement groupByArray = actual.GetProperty(QueryBuilder.GROUP_BY_FIELD_NAME);
+
+            string expected = await GetDatabaseResultAsync(msSqlQuery);
+            JsonDocument expectedDocument = JsonDocument.Parse(expected);
+            JsonElement expectedArray = expectedDocument.RootElement;
+            SqlTestHelper.AssertNumericAggregations(groupByArray, expectedArray, isfieldsPresentInResponse: false);
+        }
+
+        /// <summary>
+        /// Test to check GraphQL support for sum aggregation with aliases.
+        /// This test verifies that the SQL query results are correctly mapped to the expected GraphQL format.
+        /// </summary>
+        [TestMethod]
+        public async Task TestSupportForGroupByHavingAggregation(string msSqlQuery)
+        {
+            string graphQLQueryName = "stocks_prices";
+            string graphQLQuery = @"
+    {
+        stocks_prices {
+            groupBy(fields: [categoryid, pieceid]) {
+                aggregations {
+                    sum_price: sum(field: price, having:{ gt: 50 })
+                }
+            }
+        }
+    }";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLQuery, graphQLQueryName, isAuthenticated: false);
+            JsonElement groupByArray = actual.GetProperty(QueryBuilder.GROUP_BY_FIELD_NAME);
+
+            string expected = await GetDatabaseResultAsync(msSqlQuery);
+            JsonDocument expectedDocument = JsonDocument.Parse(expected);
+            JsonElement expectedArray = expectedDocument.RootElement;
+            SqlTestHelper.AssertNumericAggregations(groupByArray, expectedArray, isfieldsPresentInResponse: false);
+        }
+
+        /// <summary>
+        /// Test to check GraphQL support for sum aggregation with aliases.
```

Comment:
nit: summary for all the test cases above should be fixed - says sum aggregation.